### PR TITLE
Add profile store

### PR DESCRIPTION
**Pull request by mkcode92**

### CODE REVIEW

**Comment by mkcode92** [comment]
Changed: apply bakdata codestyle, shared deployment from justfile, deployment of frontend on arbitrary quick host, public images

File: profile-store/deployment/Readme.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+## Pre-Requisites
+
+* Access to a quick instance (version > 0.2.2)
```
Comment (torbsto) [inline profile-store/deployment/Readme.md]: Is 0.2.2 really correct?

File: profile-store/deployment/Readme.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+* [helm](https://helm.sh/docs/intro/install/) installed
+* [streams-bootstrap-charts](https://github.com/bakdata/streams-bootstrap/tree/master/charts) installed
+* [quick-cli](https://github.com/bakdata/quick-cli) > 0.2.2 installed
+* [just](https://github.com/casey/just)
```
Comment (torbsto) [inline profile-store/deployment/Readme.md:10]: Maybe mark optional

File: profile-store/deployment/Readme.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+## Pre-Requisites
+
+* Access to a quick instance (version > 0.2.2)
+* [helm](https://helm.sh/docs/intro/install/) installed
```
Comment (torbsto) [inline profile-store/deployment/Readme.md]:
```suggestion
* [Helm](https://helm.sh/docs/intro/install/) installed
```

File: profile-store/streams-apps/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    repositories {
+        mavenCentral()
+        maven {
+            url = uri("https://packages.confluent.io/maven/")
+        }
+    }
+}
+
+subprojects {
+    group = "com.bakdata.profilestore"
```
Comment (torbsto) [inline profile-store/streams-apps/build.gradle.kts]:
```suggestion
    group = "com.bakdata.quick.profilestore"
```

**Review by torbsto — COMMENTED** [review]
1. More documentation to give readers some orientation would be helpful, e.g.:
    - a bullet point in the root ReadMe with a short description of the project
    - a ReadMe in /profile-store/ explaining the general gist and a description for each subdirectory
2. The classes in the java project are not in a package, please move them into the package corresponding to the group. 
3. A bit of JavaDoc would be helpful, explaining what each application roughly does

File: profile-store/Readme.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# Profile store
+
+This project combines several features of Quick
+for user profiles on real-world music listening behaviour.
+The results of multiple independent apps are managed in a single schema.
+This includes a recommendation REST-service.
+The frontend also makes use of graphql subscriptions and mutations.
```
Comment (torbsto) [inline profile-store/Readme.md]:
```suggestion
The frontend also makes use of GraphQL subscriptions and mutations.
```

File: profile-store/Readme.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+* **data** a sample of the LFM-1b listening dataset 
+together with a python script that was used to extract the data.
+* **deployment** the graphql schema, the environment configuration
+ and a justfile to deploy the streams apps and the frontend in kubernetes.
+* **frontend** a python app that uses the Quick gateway as a backend.
+* **streams-apps** the source code for the kafka streams apps
```
Comment (torbsto) [inline profile-store/Readme.md]:
```suggestion
* **streams-apps** the source code for the Kafka Streams apps 
```

File: profile-store/Readme.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+This project combines several features of Quick
+for user profiles on real-world music listening behaviour.
+The results of multiple independent apps are managed in a single schema.
+This includes a recommendation REST-service.
+The frontend also makes use of graphql subscriptions and mutations.
+
+* **data** a sample of the LFM-1b listening dataset 
+together with a python script that was used to extract the data.
+* **deployment** the graphql schema, the environment configuration
```
Comment (torbsto) [inline profile-store/Readme.md]:
```suggestion
* **deployment** the GraphQL schema, the environment configuration
```

File: profile-store/Readme.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+ and a justfile to deploy the streams apps and the frontend in kubernetes.
```
Comment (torbsto) [inline profile-store/Readme.md]:
```suggestion
 and a justfile to deploy the streams apps and the frontend in Kubernetes.
```

**Pull request merged**